### PR TITLE
Solve selected pylint issues

### PR DESCRIPTION
**Pull request by berland**
Mostly by disable statements close to the offending line.
Some cleanup in unused fixtures

FIXME added to a test, this looks like a real miss-out uncovered by pylinting.

### CODE REVIEW

File: tests/jobs/localisation/test_integration.py [hunk 1]
```diff
@@ -57,6 +56,12 @@ def test_localisation(setup_ert, obs_group_add, param_group_add, expected, snaps
     snapshot.assert_match(
         str(ert.update_configuration.dict()), "localisation_configuration"
     )
+    assert [
```
Comment (berland) [inline tests/jobs/localisation/test_integration.py:59]: @oyvindeide , could you have a peek at this added `assert`, does it make sense to include, or should the argument to the test be dropped?

File: tests/jobs/design2params/test_design2params.py [hunk 2]
```diff
@@ -178,7 +178,8 @@ def test_support_multiple_formats(tmp_path, ext, engine):
     pd.testing.assert_frame_equal(df, loaded_df, check_dtype=False)
 
 
-def test_run_realization_not_exist(input_data):
+@pytest.mark.usefixtures("input_data")
+def test_run_realization_not_exist():
     with pytest.raises(SystemExit):
```
Comment (kvashchuka) [inline tests/jobs/design2params/test_design2params.py:183]: This test is passing even without fixture

File: tests/jobs/design2params/test_design2params.py [hunk 3]
```diff
@@ -190,17 +191,20 @@ def test_run_realization_not_exist(input_data):
         )
 
 
-def test_open_excel_file_not_existing(input_data):
+@pytest.mark.usefixtures("input_data")
+def test_open_excel_file_not_existing():
     with pytest.raises(SystemExit):
```
Comment (kvashchuka) [inline tests/jobs/design2params/test_design2params.py:196]: and this one

File: tests/jobs/design2params/test_design2params.py [hunk 4]
```diff
@@ -190,17 +191,20 @@ def test_run_realization_not_exist(input_data):
...
         design2params._read_excel("not_existing.xlsx", "DesignSheet01")
 
 
-def test_open_excel_file_not_xls(input_data):
+@pytest.mark.usefixtures("input_data")
+def test_open_excel_file_not_xls():
     with pytest.raises(SystemExit):
```
Comment (kvashchuka) [inline tests/jobs/design2params/test_design2params.py:202]: and this one

File: tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py [hunk 5]
```diff
@@ -72,8 +73,8 @@ def assert_homogen_clusters(config):
 
 
 def generate_simulated_responses(
-    forward_polynomials,
-    parameter_distribution,
+    _forward_polynomials,
+    _parameter_distribution,
```
Comment (kvashchuka) [inline tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py:77]: Are these two parameters even used? I tried to remove them completely and I don't think it broke anything ...

File: tests/jobs/design2params/test_design2params.py (hunk 2 above)
Comment (berland) [reply tests/jobs/design2params/test_design2params.py:183]: fixed by asserting the xls file is there.

File: tests/jobs/design2params/test_design2params.py (hunk 3 above)
Comment (berland) [reply tests/jobs/design2params/test_design2params.py:196]: fixed

File: tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py (hunk 5 above)
Comment (berland) [reply tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py:77]: They are not used in this test no, but I guessed it was nice to keep them in for illustration. The underscore just silences `pylint`.

File: tests/jobs/localisation/test_integration.py (hunk 1 above)
Comment (sondreso) [reply tests/jobs/localisation/test_integration.py:59]: The `expected` parameter has never been used in this test. I think it makes more sense to just remove it

File: tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py (hunk 5 above)
Comment (kvashchuka) [reply tests/jobs/misfit_preprocessor/test_misfit_preprocessor.py]: The test is changed such that it uses forward polynomials and parameter_distribution

**Review by pinkwah — APPROVED** [review]

**Pull request merged**